### PR TITLE
[CI] Replace Python script by distribution to build bundle

### PR DESCRIPTION
**Pull request by ThibaudAV**
**Issue**
https://github.com/gravitee-io/issues/issues/7813

**Description**

ci: refactor prepare APIM package bundles

To test run `npm run package_zip -- --version=3.18.0 --dry-run`
Result : https://app.circleci.com/pipelines/github/gravitee-io/gravitee-api-management/9754/workflows/484f24b1-4f5a-496c-805d-9640f2770598/jobs/123233

⚠️ :
The only change I noticed was on the prefix `graviteeio` inside `graviteeio-full-3.18.0.zip` has not been retained. 
but if needed I can add the code that renames these parts when composing the `graviteeio-full`. 

IMHO: I would be up for only and always having `gravitee` prefix and remove all `graviteeio` only used for the full zip ... 🤷‍♂️ 

**Additional context**

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-ccvqwgavej.chromatic.com)

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.z6.web.core.windows.net/7813-remove-python/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

### CODE REVIEW

File: release/ci-steps/package-bundles.mjs [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+console.log(chalk.blue(`Step 1: Prepare APIM distribution`));
+await within(async () => {
+  cd('../');
+  await $`mvn flatten:flatten`;
+});
+
+const pomXml = await fs.readFile(`../gravitee-apim-distribution/.flattened-pom.xml`, 'utf-8');
+const jsonPom = JSON.parse(await xml2json.toJson(pomXml, {}));
+
+const releasingVersion = argv.version; // jsonPom.project.parent.version;
```
Comment (ThibaudAV) [inline release/ci-steps/package-bundles.mjs:18]: to discuss

File: .circleci/config.yml [hunk 2]
```diff
@@ -1448,10 +1448,11 @@ jobs:
 
     package-bundle:
         docker:
-            - image: cimg/python:3.10
+            - image: cimg/openjdk:11.0-node
         resource_class: medium
         environment:
             RELEASE_VERSION: << pipeline.parameters.graviteeio_version >>
+            ARTIFACTORY_REPO_URL: "https://odbxikk7vo-artifactory.services.clever-cloud.com/external-dependencies-n-gravitee-all"
```
Comment (ThibaudAV) [inline .circleci/config.yml:1455]: 🤔  could be a composition from existing ARTIFACTORY_URL env var + `external-dependencies-n-gravitee-all`

File: .circleci/config.yml [hunk 3]
```diff
@@ -1461,108 +1462,16 @@ jobs:
                   var-name: ARTIFACTORY_PASSWORD
             - checkout
             - run:
-                  name: "Install python libraries"
-                  command: |
-                      pip install requests
-
+                  name: Install dependencies
+                  command: printenv
```
Comment (gaetanmaisse) [inline .circleci/config.yml]: To remove?

File: release/steps/2-package_zip.mjs [hunk 4]
```diff
@@ -28,7 +26,7 @@ if (tagSteps !== 'y') {
 
 // Use the preconfigured payload from config folder with the good parameters
 const body = {
-  branch: versions.branch,
+  branch: '7813-remove-python',//versions.branch,
```
Comment (gaetanmaisse) [inline release/steps/2-package_zip.mjs:29]: to remove

File: release/steps/2-package_zip.mjs [hunk 5]
```diff
@@ -14,8 +14,6 @@ console.log(chalk.blue(`Triggering Package Zip Pipeline`));
 const tagSteps = await question(
   chalk.blue(
     '⚠️ Before packaging, you have to:\n' +
-      " - Update gravitee-io/release's release.json to update 'gravitee-api-management' version\n" +
-      ' - Commit\n' +
       " - Create a tag for the new version (To create a tag, use 'git tag -a {version}' then 'git push origin {version}')\n" +
```
Comment (gaetanmaisse) [inline release/steps/2-package_zip.mjs:17]: I think we can remove this also? And just ask if the distribution pom is up to date as the packaging will be based on that. WDYT?

File: release/ci-steps/package-bundles.mjs (hunk 1 above)
Comment (gaetanmaisse) [reply release/ci-steps/package-bundles.mjs:18]: I would go for the `argv.version` for now

**Comment by gaetanmaisse** [comment]
Tested in dry-run mode -> ✅  https://gravitee-dry-releases-downloads.cellar-c2.services.clever-cloud.com/index.html#graviteeio-apim/

File: release/steps/2-package_zip.mjs (hunk 4 above)
Comment (ytvnr) [inline release/steps/2-package_zip.mjs:29]: put your PR in draft to avoid a wild merge :p

**Review by gaetanmaisse — APPROVED** [review]

**Comment by gaetanmaisse** [comment]
Hmmm it looks we are missing a folder level in the full package:

See ⬇️ , the first command was run against the current 3.18.1 bundle and the 2nd one against the one generated with this PR

image

File: .circleci/config.yml (hunk 3 above)
Comment (ThibaudAV) [reply .circleci/config.yml]: yes for testing to see with env already available

File: release/steps/2-package_zip.mjs (hunk 4 above)
Comment (ThibaudAV) [reply release/steps/2-package_zip.mjs:29]: yes normally inside dedicated commit to remove

**Comment by ThibaudAV** [comment]
I must have lost it with the last change 🤔 
i fix it soon

File: release/steps/2-package_zip.mjs (hunk 5 above)
Comment (ThibaudAV) [reply release/steps/2-package_zip.mjs:17]: yes you right. confirm by recent release i made

**Pull request closed without merging**